src/RegistryPageController.php

```diff
@@ -205,16 +207,81 @@ public function RegistryEntries($paginated = true)
         return $this->queryList($where, $orderby, $start, $this->dataRecord->getPageLength(), $paginated);
     }
 
-    public function Columns($result = null)
+    /**
+     * Loosely check if the record can be sorted by a property
+     * @param  string $property
+     * @return boolean
+     */
+    public function canSortBy($property)
+    {
+        $canSort = false;
+        $singleton = $this->dataRecord->getDataSingleton();
+
+        if ($singleton) {
+            $properties = explode('.', $property);
+
+            $relationClass = $singleton->getRelationClass($properties[0]);
+            if ($relationClass) {
+                if (count($properties) <= 2 && singleton($relationClass)->hasDatabaseField($properties[1])) {
+                    $canSort = true;
+                }
+            } elseif ($singleton instanceof DataObject) {
+                if ($singleton->hasDatabaseField($property)) {
+                    $canSort = true;
+                }
+            }
+        }
+
+        return $canSort;
+    }
+
+    /**
+     * Format a set of columns, used for headings and row data
+     * @param  ViewabledData $result The row context
+     * @return ArrayList
+     */
+    public function columns($result = null)
     {
-        $columns = $this->dataRecord->getDataSingleton()->summaryFields();
+        $singleton = $this->dataRecord->getDataSingleton();
+        $columns = $singleton->summaryFields();
         $list = ArrayList::create();
+
         foreach ($columns as $name => $title) {
+            // Check for unwanted parameters
+            if (preg_match('/[()]/', $name)) {
+                throw new RegistryException(_t(
+                    'SilverStripe\\Registry\\RegistryPageController.UNWANTEDCOLUMNPARAMETERS',
+                    "Columns do not accept parameters"
+                ));
+            }
+
+            // Get dot deliniated properties
+            $properties = explode('.', $name);
+
+            // Increment properties for value
+            $context = $result;
+            foreach ($properties as $property) {
+                if ($context instanceof ViewableData) {
+                    $context = $context->obj($property);
+                }
+            }
+
+            // Check for link
+            $link = null;
+            $useLink = $singleton->config()->get('use_link');
+            if ($useLink !== false) {
+                if ($result && $result->hasMethod('link')) {
+                    $link = $result->Link();
+                }
+            }
+
+            // Format column
             $list->push(ArrayData::create([
                 'Name' => $name,
                 'Title' => $title,
-                'Link' => (($result && $result->hasMethod('Link')) ? $result->Link() : ''),
-                'Value' => ($result ? $result->obj($name) : '')
+                'Link' => $link,
+                'Value' => isset($context) ? $context : null,
```

Comment:
`$content` is always set (you assign the value of `$result` to it) so you could remove this and just use the variable